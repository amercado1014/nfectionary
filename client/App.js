import React from 'react';
import { StyleSheet, Text, View, TouchableOpacity, Image, Picker, Select } from 'react-native';
import { Dropdown } from 'react-native-material-dropdown';

export default class App extends React.Component {
  constructor(){
    super();
    this.state = {
      response: '',
      latitude: null,
      longitude: null,
      state: '',
      error: null
    }
  }

  componentDidMount() {
    this.callApi()
      .then(res => this.setState({ response: res.express }))
      .catch(err => console.log(err));
  }

  callApi = async () => {
    const response = await fetch('http://localhost:5000/api/hello');
    const body = await response.json();
    if (response.status !== 200) throw Error(body.message);
    return body;
  };

  currentLocation = async () => {
   const location = await navigator.geolocation.getCurrentPosition(
      async (position) => {
        const {latitude, longitude} = position.coords;
        console.log(latitude, longitude);
        const myApi = 'AIzaSyCrg4u9bjIN_fZzCQNKmD2HN3iVQLKigXk';
        const response = await fetch('https://maps.googleapis.com/maps/api/geocode/json?address=' + latitude + ',' + longitude + '&key=' + myApi)
        const location = await response.json()
        const { results } = location;
        if (results[1]) {
					for (var i = 0; i < results.length; i++) {
						if (results[i].types[0] === "locality") {
							var city = results[i].address_components[0].short_name;
							var state = results[i].address_components[2].short_name;
              console.log(city, state);
						}
					}
				}
      },
      (error) => this.setState({ error: error.message }),
      { enableHighAccuracy: true, timeout: 20000, maximumAge: 1000 },
    );
  }

  render() {
    return (
      <View style={styles.container}>
        <Text>Welcome to NFectinoary</Text>
        <Text>Select a state or use your location</Text>
        <TouchableOpacity 
          onPress={this.currentLocation}
          title="Current Location"
          style={styles.button}
          accessibilityLabel="Use your current location"
          ><Text> Find Location </Text>
          </TouchableOpacity>
          {/* <Dropdown
            label='Favorite Fruit'
            data={data}
          /> */}
          {/* <Picker
            selectedValue={this.state.state}
            style={{ height: 50, width: 200 }}
            onValueChange={(itemValue, itemIndex) => this.setState({state: itemValue})}>
            <Picker.Item label="Colorado" value="Colorado" />
            <Picker.Item label="California" value="California" />
            <Picker.Item label="Alabama" value="Alabama" />
            <Picker.Item label="New Mexico" value="New Mexico" />
          </Picker> */}
      </View>
    );
  }
}

const data = [{
  value: 'Colorado',
}, {
  value: 'California',
}, {
  value: 'Alabama',
},
{
  value: 'New Mexico',
}
];


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  button: {
    alignItems: 'center',
    backgroundColor: '#DDDDDD',
    padding: 10
  }
});
