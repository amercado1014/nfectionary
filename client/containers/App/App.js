import React, { Component } from 'react';
import { createStackNavigator } from 'react-navigation';
import HomeScreen from '../HomeScreen/HomeScreen';
import StateDisplay from '../StateDisplay/StateDisplay';
import DiseaseDisplay from '../DiseaseDisplay/DiseaseDisplay';

const RootStack = createStackNavigator(
  {
    Home: HomeScreen,
    StateDisplay: StateDisplay,
    DiseaseDisplay: DiseaseDisplay
  },
  {
    initialRouteName: 'Home',
  }
);

export default class App extends Component {
  render() {
    return <RootStack />;
  }
}