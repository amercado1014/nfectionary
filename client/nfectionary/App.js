import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

export default class App extends React.Component {
  constructor() {
    super();
    this.state = {
      response: ''
    };
  }
  

  componentDidMount() {
    this.callApi()
      .then(res => this.setState({ response: res.express }))
      .catch(err => console.log(err));
  }

  callApi = async () => {
    console.log('in the call api');
    
    const response = await fetch('http://localhost:5000/api/hello');
    console.log(response);
    
    const body = await response.json();
    console.log(body);
    
    if (response.status !== 200) throw Error(body.message);

    return body;
  };
  render() {
    //console.log(this.state);
    
    return (
      <View style={styles.container}>
        <Text>TEST on your app!</Text>
        <Text>Changes you make will automatically reload.</Text>
        <Text>{this.state.response}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
