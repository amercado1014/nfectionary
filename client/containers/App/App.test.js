import React from 'react';
import App from './App';
// import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('App', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
