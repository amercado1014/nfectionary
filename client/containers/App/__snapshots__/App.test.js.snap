// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#E9E9EF",
            "bottom": 0,
            "left": 0,
            "marginTop": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Welcome to NFectionary
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Select a state or use your location
          </Text>
          <View
            accessibilityLabel="Use your current location"
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#DDDDDD",
                "opacity": 1,
                "padding": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
               Get Started 
            </Text>
          </View>
          <View
            animated={true}
            defaultIndex={-1}
            defaultValue="Select State"
            disabled={false}
            keyboardShouldPersistTaps="never"
            onSelect={[Function]}
            options={Array []}
            scrollEnabled={true}
            showsVerticalScrollIndicator={true}
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "fontSize": 12,
                      },
                      undefined,
                    ]
                  }
                >
                  Select State
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "backgroundColor": "#F7F7F7",
            "borderBottomColor": "#A7A7AA",
            "borderBottomWidth": 0.5,
            "height": 64,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 20,
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "flexDirection": "row",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                accessibilityTraits="header"
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, .9)",
                    "fontSize": 17,
                    "fontWeight": "700",
                    "marginHorizontal": 16,
                    "textAlign": "center",
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
